File: abc/207/b_ans.py
def main():
    # input
    A, B, C, D = map(int, input().split())

    # compute
    ans = -1
    diff = C*D - B
    if diff > 0:
        ans = (A+diff-1) // diff

    # output
    print(ans)


if __name__ == '__main__':
    main()

File: abc/177/b_ans.c
#include <stdio.h>
#include <string.h>

int mymin(int a, int b);

int main(){
  // input
  char S[1000]; scanf("%s", S);
  int s = strlen(S);
  char T[s]; scanf("%s", T);
  int t = strlen(T);

  //calculate
  int ans = t;
  int i, j;
  for (i = 0; i < s-t+1; ++i){
    int diff = 0;
    for (j = 0; j < t; ++j){
      if (T[j] != S[i+j]){
        diff++;
      }
    }
    ans = mymin(ans, diff);
  }

  // output
  printf("%d", ans);
  return 0;
}

// find min
int mymin(int a, int b){
  if (a >= b){
    return b;
  }
  else{
    return a;
  }
}